resources/fetchersManager/data.go

```diff
@@ -57,19 +58,21 @@ func NewData(log *logp.Logger, interval time.Duration, timeout time.Duration, fe
 
 // Run starts all configured fetchers to collect resources.
 func (d *Data) Run(ctx context.Context) Stop {
-	go d.fetchAndSleep(ctx)
+	fetchCtx, cancel := context.WithCancel(ctx)
+	go d.fetchAndSleep(fetchCtx, cancel)
```

Comment:
lifted the context, making it used by fetchers and also the grace period timeout context. 

so cloudbeat will shutdown once grace period timeout context is done, which will happen:
- once the timeout has exceeded, or 
- when its parent context was canceled

whichever comes first. 

to get an estimate what will come first, we need to look at where we return from `fetchAndSleep`, as that is when the context is canceled.

we return in 3 places: 
https://github.com/elastic/cloudbeat/blob/c1d27fee3ebfc69cbf85c9a8570041e413f4c64b/resources/fetchersManager/data.go#L85-L93
 
 - stop won't get called until the grace period context is done , won't run
 - the context won't be done until we exit fetchAndSleep, won't run (can be deleted?)
 - when the next interval will run after already receiving a notice, it will return and then cancel the shared context 
 
 so the shorter the fetch period and the higher grace period, the more likely we are to exit **before** the grace period ended.